cart-service/src/lib/dovetech-response-mapper.ts

```diff
@@ -245,21 +244,16 @@ const getDirectDiscountShippingAction = (
   return shippingDiscount;
 };
 
-const buildSetCustomTypeAction = (
+const buildSetCustomTypeActions = (
   dtResponse: DoveTechDiscountsResponse,
   couponCodeAcceptedActions: CouponCodeAcceptedAction[],
   currencyCode: string,
   commitId: string | null = null
 ) => {
-  const couponCodes: CouponCode[] = couponCodeAcceptedActions.map((a) => ({
-    code: a.code,
-  }));
-
-  const serialisedCouponCodes = JSON.stringify(couponCodes);
-
-  const fields: { [key: string]: string } = {
+  // set metadata action
+  const fields: { [key: string]: string | string[] } = {
```

Comment:
This looks like the type for the fields:

```suggestion
  const fields: FieldContainer = {
```

Reply:
changed